sdf.md

```diff
@@ -829,10 +827,117 @@ numbers can be used.
+| enum          | array of strings            | abbreviation for string-valued named alternatives                   | N/A     |
 {: #sdfdataqual2 title="SDF-defined Qualities of sdfProperty and sdfData"}
 
 (3) note that the quality `unit` was called `units` in SDF 1.0.
 
+### sdfChoice
+
+Data can be a choice of named alternatives, called `sdfChoice`.
+Each alternative is identified by a name (string, key in the JSON
+object used to represent the choice) and a set of dataqualities
+(object, the value in the JSON object used to represent the choice).
+
+sdfChoice merges the functions of two constructs found in {{-jso}}:
+
+* `enum`
+
+  What would have been
+
+  ~~~ json
+  "enum": ["foo", "bar", "baz"]
+  ~~~
+
+  in SDF 1.0, is often best represented as:
+
+  ~~~ json
+  "sdfChoice": {
+    "foo": { "description": "This is a foonly"},
+    "bar": { "description": "As defined in the second world congress"},
+    "baz": { "description": "From zigbee foobaz"}
+  }
+  ~~~
+
+  This allows the placement of other dataqualities such as
+  `description` in the example.
+
+  If an enum needs to use a data type different from text string,
+  e.g. what would have been
+
+  ~~~ json
+  "type": "number",
+  "enum": [1, 2, 3]
+  ~~~
+
+  in SDF 1.0, is represented as:
+
+  ~~~ json
+  "type": "number",
+  "sdfChoice": {
+    "a-better-name-for-alternative-1": { "const": 1 },
+    "alternative-2": { "const": 2 },
+    "the-third-alternative": { "const": 3 }
+  }
+  ~~~
+
+  where the string names obviously would be chosen in a way that is
+  descriptive for what these numbers actually stand for; sdfChoice
+  also makes it easy to add number ranges into the mix.
+
+  (Note that `const` can also be used for strings as in the previous
+  example, e.g., if the actual string value is indeed a crucial
+  element for the data model.)
+
+* anyOf
+
+  {{-jso}} provides a type union called `anyOf`, which provides a
+  choice between anonymous alternatives.
+
+  What could have been
+
+  ~~~ json
+  "anyOf": [
+    {"type": "array", "minItems": 3, "maxItems": "3", "items": {
+       "sdRef": "rgbVal"}}
+    {"type": "array", "minItems": 4, "maxItems": "4", "items": {
+       "sdRef": "cmykVal"}}
+  ]
+  ~~~
+
+  in {{-jso}} can be more descriptively notated in SDF as:
+
+  ~~~ json
+  "sdfChoice": {
+    "rgb": {"type": "array", "minItems": 3, "maxItems": "3", "items": {
+              "sdRef": "rgbVal"}}
+    "cmyk": {"type": "array", "minItems": 4, "maxItems": "4", "items": {
+              "sdRef": "cmykVal"}}
+  ]
+  ~~~
+
+Note that there is no need in SDF for the type intersection construct
+`allOf` or the peculiar type-xor construct `oneOf` found in {{-jso}}.
+
+As a simplification for readers of SDF specifications accustomed to
+the {{-jso}} enum keyword, this is retained, but limited to a choice
+of text string values, such that
+
+~~~ json
+"enum": ["foo", "bar", "baz"]
+~~~
+
+is syntactic sugar for
+
+~~~ json
+"sdfChoice": {
+  "foo": { "const": "foo"},
```

Comment:
Would `"foo": {}` be also OK here?

Reply:
That is the little conflict between the information model character of SDF and the fact that we are expressing this in JSON instance.  By putting in the constant "foo", there is only one instance that can stand for this choice, and that is consistent with the meaning of jso's enum. The fact that a protocol binding could use the number 4711 instead doesn't really belong here; this should be done in the protocol binding.

Reply:
I would expect {"const": "foo") to be a popular string binding for this element, but not included in the adopted model